### ISSUE TITLE
Unable to use `reuseport` with `gen_udp:open`

### ISSUE THREAD

**Issue by LVala** [erlang/otp#7569]
**Describe the bug**
Documentation suggests that from OTP 26, `inet:setopts` allows to set the `reuseport`. When trying to set the option with `gen_udp:open`, I get the following error:

```erl
Erlang/OTP 26 [erts-14.0.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Eshell V14.0.2 (press Ctrl+G to abort, type help(). for help)
1> gen_udp:open(55001, [{reuseport, true}]).
** exception exit: badarg
     in function  inet_udp:open/2 (inet_udp.erl, line 56)
```

After looking at `otp/lib/kernel/src/inet.erl` I noticed that function `udp_options` doesn't list `reuseport` as an available option, which is not the case for e.g. `listen_options` and indeed, `gen_tcp:listen` works fine with the `reuseport`:

```erl
1> {ok, Socket} = gen_tcp:listen(55001, [{reuseport, true}]).
{ok,#Port<0.3>}
```

Getting back to UDP, the option can be set via `inet:setopts`, but that doesn't help, as the next opened socket need to use `reuseport` as well, which must be set before binding to address:

```erl
1> {ok, Socket} = gen_udp:open(55001).
{ok,#Port<0.3>}
2> ok = inet:setopts(Socket, [{reuseport, true}]).
ok
3> inet:getopts(Socket, [reuseport]).
{ok,[{reuseport,true}]}
4> {ok, Sock2} = gen_udp:open(55001).
** exception error: no match of right hand side value {error,eaddrinuse}
```

My workaround was to use `socket` module, which works as expected, but the above does not seem to be the desired behaviour.

**To Reproduce**
Run `gen_udp:open(55001, [{reuseport, true}]).`.

**Expected behavior**
I expect to be able to use `reuseport` option when opening UDP socket via `gen_udp:open`.

**Affected versions**
OTP 26.0.2
MacOS M1
```console
$ sysctl kern.version
kern.version: Darwin Kernel Version 22.3.0: Thu Jan  5 20:48:54 PST 2023; root:xnu-8792.81.2~2/RELEASE_ARM64_T6000
```

**Additional context**
None.

**Comment by RaimoNiskanen** [comment]
This seems to be a bug in PR #6522 that missed to allow the new options for `gen_udp:open/1,2`.
I'll start on a fix.

**Comment by RaimoNiskanen** [comment]
Fix merged to 'maint' to be released in OTP-26.1.
Thank you for reporting this issue!
